controllers/limitador_controller.go

```diff
@@ -129,9 +132,58 @@ func (r *LimitadorReconciler) reconcileSpec(ctx context.Context, limitadorObj *l
 	if err := r.reconcilePdb(ctx, limitadorObj); err != nil {
 		return ctrl.Result{}, err
 	}
+
+	if err := r.reconcilePodLimitsHashAnnotation(ctx, limitadorObj); err != nil {
+		return ctrl.Result{}, err
+	}
+
 	return ctrl.Result{}, nil
 }
 
+func (r *LimitadorReconciler) reconcilePodLimitsHashAnnotation(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) error {
+	podList := &v1.PodList{}
+	options := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(limitador.Labels(limitadorObj)),
+		Namespace:     limitadorObj.Namespace,
+	}
+	if err := r.Client().List(ctx, podList, options); err != nil {
+		return err
+	}
+
+	if len(podList.Items) == 0 {
+		return errors.New("no limitador pods founds")
+	}
+
+	// Replicas won't change if spec.Replicas goes from value to nil
+	if limitadorObj.Spec.Replicas != nil && len(podList.Items) != int(limitadorObj.GetReplicas()) {
+		return errors.New("unexpected number of limitador pods")
+	}
+
+	// Calculate hash once for all pods
+	hash, err := limitadorObj.LimitsHash()
```

Comment:
This is what I would have done otherwise, hence I would like to discuss. Not saying it is wrong like it is, but I think it is less robust comparing to what I would have done. I would have added pod annotation with the `resourceVersion` of the `ConfigMap`. You would need to sync reconciliation of the config map and then pods' annotation, i.e. first config map, and when available, check pods annotations.

Reply:
Oh cool, using `resourceVersion` of the `Config` map instead is a good idea. I didn't think of that 👍